### DISCUSSION TITLE
How to update fontsize and width of textbox on object:scaling event?

### DISCUSSION THREAD

**Discussion by niketaOptimumnbrew** [fabricjs/fabric.js#9832] [Q&A]
Hi guys , I'm using fabric.js version 5.3.0

I have one question that how to manage fontsize and width of textbox on fabric.js textbox scale event?

**Comment by ShaMan123** [comment]
Don't use the scaling event.
Create your own control handler.
Refer to the changeWidth function as a starting point.

**Reply by niketaOptimumnbrew** [reply]
@ShaMan123 changeWidth  will update width horizontally not the fontsize , I want to update according to scale value on object:scaling event.

**Reply by ShaMan123** [reply]
I understand that.
I suggested you use that logic to create a custom handler.

**Reply by niketaOptimumnbrew** [reply]
@ShaMan123 Can't be done without using the custom handler?

**Reply by ShaMan123** [reply]
Do as you please.
Hopefully you will find the easy and robust way.

**Reply by niketaOptimumnbrew** [reply]
@ShaMan123 Thank you for your suggestion , will try with custom handler.

**Comment by asturur** [comment]
The issue with doing on scale event is that the scale event happens after scale is done.
So you will end up reassign the previous scale ( that would be 1 most of the time, depending on your logic ) and then calculate the new width and height and font size and apply them, you have to keep track of your previous scale value too if is not 1.

Is done as any other logic you want to run on a scale event, is not clear if you are asking to write the logic for you that is a bare multiplication or something else.

Both method are valid, the control api offers you a way to go more custom, you can use the one that  reach your goal.
But fabricJS is nearly 14 years old, this question has been asked many times and answered many times, i  m sure if you search here or stack overflow you find it done.

**Reply by niketaOptimumnbrew** [reply]
Thank you  for answering my question :)
will try with custom handler and let's see what outcome it gives.
